mount/mount_linux.go

```diff
@@ -112,6 +112,9 @@ func unmount(target string, flags int) error {
 // are no mounts remaining (EINVAL is returned by mount), which is
 // useful for undoing a stack of mounts on the same mount point.
 func UnmountAll(mount string, flags int) error {
+	if _, err := os.Stat(mount); err != nil {
+		return nil
```

Comment:
`return err`? or needs `os.IsNotExist` check?

Reply:
We thought the same, but when you do `os.Stat("")` it returns an `EINVAL` error [as you can see](https://play.golang.org/p/wEuJ0wn9wHS). We want to skip the unmount in 2 cases - if it is an invalid path (`""`) or if it does not exist on disk. That's why we do not do the `os.IsNotExist` check. Our thinking was that If you cannot stat the directory it is doubtful if you will be able to unmount it anyway.

If you prefer we can have 2 checks instead - a check for an empty string and an `os.IsNotExist` check. What do you think?

Reply:
+1 for having 2 checks.
The code should also have comment lines about this.

cc @dmcgowan @fuweid

Reply:
we can check the rootfs in the shim, not in mount package. mount package don't need to carry this logic.